From: itzexor <itzexor@gmail.com>
Date: Sun, 12 Jun 2016 15:08:44 -0700
Subject: pathbar: remove hardcoded width for sliders

from https://github.com/GNOME/nautilus/commit/6d87f24a8bc6bd5c376c0cb851fb24e440d766f6
---
 src/nemo-pathbar.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/nemo-pathbar.c b/src/nemo-pathbar.c
index a6c76cb..11bd09a 100644
--- a/src/nemo-pathbar.c
+++ b/src/nemo-pathbar.c
@@ -491,7 +491,11 @@ nemo_path_bar_get_preferred_width (GtkWidget *widget,
         *minimum = MAX (*minimum, child_min);
         *natural = MAX (*natural, child_nat);
     }
-    path_bar->priv->slider_width = MIN (height * 2 / 3 + 5, height);
+
+    gtk_widget_get_preferred_width (path_bar->priv->down_slider_button,
+                                    &path_bar->priv->slider_width,
+                                    NULL);
+
     *minimum += path_bar->priv->slider_width * 2;
     *natural += path_bar->priv->slider_width * 2;
 }
